cfme/storage/object_store_container.py

```diff
@@ -0,0 +1,166 @@
+    """The toolbar on the Object Store Containers detail page"""
+    policy = Dropdown('Policy')
+    download = Button(title='Download summary in PDF format')
+
+
+class ObjectStoreContainerDetailsEntities(View):
+    """The entities on the Object Store Containers detail page"""
+    breadcrumb = BreadCrumb()
+    properties = SummaryTable('Properties')
+    relationships = SummaryTable('Relationships')
+    smart_management = SummaryTable('Smart Management')
+
+
+class ObjectStoreContainerDetailsAccordion(View):
+    """The accordion on the Object Store Containers details page"""
+    @View.nested
+    class properties(Accordion):  # noqa
+        tree = ManageIQTree()
+
+    @View.nested
+    class relationships(Accordion):  # noqa
+        tree = ManageIQTree()
+
+
+class ObjectStoreContainerView(BaseLoggedInPage):
+    """A base view for all the Object Store Containers pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+    flash = FlashMessages(
+        './/div[@id="flash_msg_div"]/div[@id="flash_text_div" or '
+        'contains(@class, "flash_text_div")]')
+
+    @property
+    def in_container(self):
+        return (
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Storage', 'Object Storage',
+                                                   'Object Store Containers'])
+
+
+class ObjectStoreContainerAllView(ObjectStoreContainerView):
+    """The all Object Store Containers page"""
+    toolbar = View.nested(ObjectStoreContainerToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_container and
+            self.title.text == 'Cloud Object Store Containers')
+
+
+class ObjectStoreContainerDetailsView(ObjectStoreContainerView):
+    """The detail Object Store containers page"""
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].key)
+
+        return (
+            self.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(ObjectStoreContainerDetailsToolbar)
+    sidebar = View.nested(ObjectStoreContainerDetailsAccordion)
+    entities = View.nested(ObjectStoreContainerDetailsEntities)
+
+
+class ObjectStoreContainerCollection(BaseCollection):
+    """Collection object for :py:class:'cfme.storage.object_store_container.ObjectStoreContainer'
+    """
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+
+    def instantiate(self, key, provider):
+        return ObjectStoreContainer(self, key, provider)
+
+    def all(self, provider):
+        """returning all containers objects"""
+        view = navigate_to(self, 'All')
+        containers = [self.instantiate(key=item.name, provider=provider)
+                 for item in view.entities.get_all()]
```

Comment:
REQUIRED: please use all_entity_names instead of get_all(). Otherwise, you will have to add surf_page=True, etc.